models/demos/wormhole/mistral7b/demo/demo_with_prefill.py

```diff
@@ -0,0 +1,359 @@
+    assert min_prompt_len <= max_prompt_len, f"Minimum prompt length {min_prompt_len} exceeds max len {max_prompt_len}"
+
+    if min_prompt_len < 128:
+        prefill_seq_len = 0  # For short prompts do decode-as-prefill instead
+    else:
+        prefill_seq_len = (
+            1024 if min_prompt_len > 1024 else (512 if min_prompt_len > 512 else 128)
+        )  # TODO Only supports prefill lengths of 128, 512, 1024
+        # Initial prefill tensor full of pad tokens
+        input_tokens_prefill = torch.full((len(input_prompts), prefill_seq_len), tokenizer.pad_id, dtype=torch.int32)
+
+    # Initial decode tensor full of pad tokens
+    input_tokens_decode = torch.full(
+        (len(input_prompts), max_prompt_len - prefill_seq_len), tokenizer.pad_id, dtype=torch.long
+    )
+
+    for i, encoded in enumerate(encoded_prompts):
+        if prefill_seq_len > 0:
+            input_tokens_prefill[i] = torch.tensor(encoded[:prefill_seq_len]).to(input_tokens_prefill)
+        input_tokens_decode[i, : len(encoded[prefill_seq_len:])] = torch.tensor(encoded[prefill_seq_len:]).to(
+            input_tokens_decode
+        )
+
+    input_mask = (input_tokens_decode != tokenizer.pad_id).to(torch.bool)
+
+    num_users = len(encoded_prompts)
+    logger.info(f"# of users: {num_users}")
+
+    # Select the first token from the prompts for initial decoding
+    pt_tokenized_inputs_decode = torch.tensor(input_tokens_decode)
+    pt_tokenized_inputs_prefill = torch.tensor(input_tokens_prefill)
+    emb_inputs_decode = embd(pt_tokenized_inputs_decode[:, 0]).view(model_args.max_batch_size, 1, -1)
+    if prefill_seq_len > 0:
+        emb_prefill_inputs = [
+            embd(pt_tokenized_inputs_prefill[b, :]).view(1, prefill_seq_len, -1)
+            for b in range(model_args.max_batch_size)
+        ]
+    else:
+        emb_prefill_inputs = None
+
+    # Return the rotational embedding matrix on device
+    cos, sin = precompute_freqs(model_args.head_dim, model_args.max_seq_len * 2)
+    rot_emb_matrix = freqs_to_rotation_matrix(cos, sin)
+
+    rot_emb_matrix_list = []
+    for i in range(rot_emb_matrix.shape[0]):
+        rot_emb_matrix_list.append(
+            ttnn.from_torch(
+                rot_emb_matrix[i, :, :].unsqueeze(0).unsqueeze(0), device=device, dtype=dtype, layout=ttnn.TILE_LAYOUT
+            )
+        )  # ttnn.bfloat16
+
+    return (
+        emb_inputs_decode,
+        pt_tokenized_inputs_decode,
+        emb_prefill_inputs,
+        input_mask,
+        rot_emb_matrix_list,
+        prefill_seq_len,
+        encoded_prompts,
+    )
+
+
+def run_mistral_demo(user_input, batch_size, device, instruct_mode, is_ci_env):
+    # Set Mistral flags for CI
+    if is_ci_env and instruct_mode:  # Update paths for instruct mode, otherwise use default paths for general weights
+        os.environ["MISTRAL_CKPT_DIR"] = "/mnt/MLPerf/tt_dnn-models/Mistral/mistral-7B-v0.1/instruct/"
```

Comment:
We'll still need to open an issue to centralize these directories, but that isn't blocking this PR